src/components/ContextMenuItem.js

```diff
@@ -45,44 +48,26 @@ const defaultProps = {
 class ContextMenuItem extends Component {
     constructor(props) {
         super(props);
-        this.state = {
-            success: false,
-        };
-        this.triggerPressAndUpdateSuccess = this.triggerPressAndUpdateSuccess.bind(this);
-    }
 
-    componentWillUnmount() {
-        if (!this.successResetTimer) {
-            return;
-        }
-
-        clearTimeout(this.successResetTimer);
+        this.triggerPressAndUpdateSuccess = this.triggerPressAndUpdateSuccess.bind(this);
     }
 
-    /**
-     * Called on button press and mark the run
-     */
     triggerPressAndUpdateSuccess() {
-        if (this.state.success) {
+        if (this.props.isDelayButtonStateComplete) {
             return;
         }
         this.props.onPress();
 
         // We only set the success state when we have icon or text to represent the success state
         // We may want to replace this check by checking the Result from OnPress Callback in future.
         if (this.props.successIcon || this.props.successText) {
-            this.setState({
-                success: true,
-            });
-            if (this.props.autoReset) {
-                this.successResetTimer = setTimeout(() => this.setState({success: false}), 1800);
-            }
+            this.props.toggleDelayButtonState(this.props.autoReset);
```

Comment:
In context menu, we are never using the delay. So we should just use a state instead of the HOC.

Reply:
Thanks! I’ll apply the patch and try to take a look at the rest.

Reply:
Cool, yeah I missed `CommunicationsLink`.